### PR TITLE
Add data property to `document`.

### PR DESCRIPTION
**Pull request by AlbertHilb**
Add data property to `document`. It is useful, for example, to store document level metadata: creation date, author, keywords, etc.

Of course, I'm open to any change you think needed.

### CODE REVIEW

File: src/transforms/apply-operation.js [hunk 1]
```diff
@@ -32,7 +32,8 @@ const OPERATIONS = {
   set_node: setNode,
   split_node: splitNode,
   // Selection operations.
-  set_selection: setSelection
+  set_selection: setSelection,
+  set_document_data: setDocumentData
```
Comment (ianstormtaylor) [inline src/transforms/apply-operation.js]: Does the `set_node` operation now work here already for the document? I think it should, or we should make it work.

Comment (AlbertHilb) [inline src/transforms/apply-operation.js]: No, it does not! But it can be made working with little effort.

**Comment by AlbertHilb** [comment]
Now `setNode` works also with document node and `setDocumentDataOperation` uses it internally.
Or does you prefer to eliminate `setDocumentDataOperation` too?

**Comment by AlbertHilb** [comment]
With the last commit also `setNodeByKey` does work.

**Comment by ianstormtaylor** [comment]
@AlbertHilb awesome! Yeah let's get rid of `setDocumentDataOperation` and then I think we're good to merge! Thank you!

**Comment by AlbertHilb** [comment]
I removed `setDocumentDataOperation`. :blush:

**Comment by ianstormtaylor** [comment]
Awesome, thank you for this @AlbertHilb!

**Pull request merged**